client/src/components/VideosList.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from "react";
+import Video from "./Video";
+import AddVideo from "./AddVideo";
+import "../index.css";
+import Search from "./Search";
+
+function VideosList() {
+  const [videos, setVideos] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [searchResult, setSearchResult] = useState([]);
+
+  useEffect(() => {
+    fetch("https://node-server-full-stack.onrender.com/videos")
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
+        setVideos(data);
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.error("Error fetching data:", error);
+      });
+  }, []);
+
+  const newVideo = (newInput) => {
+    newInput.id = videos.length + 1;
+    newInput.rating = 0;
+    fetch("https://node-server-full-stack.onrender.com/videos", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/JSON",
+      },
+      body: JSON.stringify(newInput),
+    })
+      .then((res) => {
+        fetchVideos();
+        return res.json();
+      })
+      .then((data) => {
+        if (data && data.message === "Video stored successfully") {
+          setVideos([...videos, newInput]);
+        }
+      })
+      .catch((error) => {
+        console.error("Error:", error);
+        alert("Something went wrong");
+      });
+  };
+
+  const deleteVideoItem = (videoId) => {
+    fetch(`https://node-server-full-stack.onrender.com/videos/${videoId}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/JSON",
+      },
+    }).then((res) => {
+      if (!res.ok) {
+          throw new Error('Network response was not ok');
+      }
+      const videoDeleted = videos.filter((el) => el.id !== videoId);
+      setVideos(videoDeleted);
+  })
+  };
+
+  const fetchVideos = () => {
+    fetch("https://node-server-full-stack.onrender.com/videos")
+      .then((res) => res.json())
+      .then((data) => {
+        setVideos(data);
+        searchResult(data);
```

Comment:
Here you're fetching the data and then you
1. `call res.json()`
2. call `setVideos` with the data
3. and then call `searchResult` -> can you explain the issue here?

Reply:
the issue here is I used searchResult , so I have to change in to setSearchResult()